### PR TITLE
NCSD-817 / 843

### PR DESCRIPTION
**Pull request by OutKa5t**
Added vNet building block to shared template
Fixed missing parameter values
Moved all parameters to parameter file
Created test parameter file for deployment outside of pipeline
Added all toenised parameter values
Added conditions to APIM to allow non APIM deployments.
Updated readme.

### CODE REVIEW

File: Resources/template.json [hunk 1]
```diff
@@ -168,6 +186,32 @@
     "deployAse": "[not(equals(parameters('aseDomainName'), ''))]"
   },
   "resources": [
+    {
+      "type": "Microsoft.Resources/deployments",
+      "name": "sharedManagedInstanceEnvironment",
+      "apiVersion": "2017-05-10",
```
Comment (rivoric) [inline Resources/template.json]: Assuming the vnet is only for DEV/SIT envs, should this not have "condition": "[equals(parameters('envType'),'dev')]", line added

File: Resources/parameters.json [hunk 2]
```diff
@@ -3,44 +3,95 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "sharedPrefix": {
-      "value": "dfc-foo-shared"
+      "value": "__ApimExists__"
```
Comment (rivoric) [inline Resources/parameters.json]: Should be __sharedPrefix__

Comment (OutKa5t) [reply Resources/parameters.json]: Good spot.

File: Resources/template.json [hunk 3]
```diff
@@ -151,6 +144,31 @@
...
+      }
+    },
+    "virtualNetworkAddressPrefix": {
+        "type": "string",
+        "defaultValue": "10.0",
+        "metadata": {
+            "description": "First 2 octects of VNet Private IP address range (VNet prefix)"
+        }
+    },
+    "virtualNetworkRoutedSubnets": {
```
Comment (rivoric) [inline Resources/template.json]: Just checking, how do you specify an empty array in Azure DevOps?

**Pull request merged**